### PR TITLE
[NBS Docs] Migrate auth0 provider to new backend system

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

Closes: https://github.com/backstage/backstage/issues/26408

Migrate the `Auth0` provider to a new auth module and update its documentation to use the new Backend System.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks @camilaibs, just a few comments 👍

File: .changeset/fair-seals-burn.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-auth-node': minor
```
Comment (Rugvip) [inline .changeset/fair-seals-burn.md:2]:
```suggestion
'@backstage/plugin-auth-node': patch
```

Not a breaking change and 0.x

**Review by Rugvip — COMMENTED** [review]
Nice! 🎉 👍 

Just a few minor bits

File: plugins/auth-backend-module-auth0-provider/api-report.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  clientID: string;
+  // (undocumented)
+  clientSecret: string;
+  // (undocumented)
+  domain: string;
+  // (undocumented)
+  passReqToCallback: true;
+  // (undocumented)
+  store: StateStore;
+}
```
[the remark this answers is not in the document]
Comment (camilaibs) [reply plugins/auth-backend-module-auth0-provider/api-report.md]: Oh, you're right! Just removed it.

File: plugins/auth-backend-module-auth0-provider/src/strategy.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Auth0InternalStrategy from 'passport-auth0';
+import { StateStore } from 'passport-oauth2';
```
[the remark this answers is not in the document]
Comment (camilaibs) [reply plugins/auth-backend-module-auth0-provider/src/strategy.ts]: Fixed 🙏🏻

**Review by Rugvip — APPROVED** [review]
🎉

File: plugins/auth-backend-module-auth0-provider/package.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    "postpack": "backstage-cli package postpack",
+    "start": "backstage-cli package start",
+    "test": "backstage-cli package test"
+  },
+  "dependencies": {
+    "@backstage/backend-plugin-api": "workspace:^",
+    "@backstage/plugin-auth-node": "workspace:^",
+    "express": "^4.17.1",
+    "passport-auth0": "^1.4.3",
+    "passport-oauth2": "^1.6.1"
```
[the remark this answers is not in the document]
Comment (camilaibs) [reply plugins/auth-backend-module-auth0-provider/package.json:41]: No because of this:
image

**Review by awanlin — APPROVED** [review]

File: plugins/auth-backend/src/providers/auth0/provider.ts
[the remark this answers is not in the document]
Comment (awanlin) [reply plugins/auth-backend/src/providers/auth0/provider.ts:1]: Looks like this is good then, thanks 👍

**Pull request merged**